### PR TITLE
added new SDK and Tests for Workloads v2022-11-01-preview

### PR DESCRIPTION
**Pull request by gargankit-microsoft**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/workloads/Azure.ResourceManager.Workloads/src/Generated/SapMonitorData.cs [hunk 1]
```diff
@@ -77,5 +87,7 @@ public string ManagedResourceGroupName
         public ResourceIdentifier MonitorSubnetId { get; set; }
         /// <summary> The ARM ID of the MSI used for SAP monitoring. </summary>
         public ResourceIdentifier MsiArmId { get; }
+        /// <summary> The ARM ID of the Storage account used for SAP monitoring. </summary>
+        public string StorageAccountArmId { get; }
```
Comment (archerzz) [inline sdk/workloads/Azure.ResourceManager.Workloads/src/Generated/SapMonitorData.cs:91]: What's the format of this property value?
- If it's a GUID, use type [Guid](https://learn.microsoft.com/en-us/dotnet/api/system.guid?view=net-6.0). See example [here](https://apiview.dev/Assemblies/Review/25a98e223bda4f39b4ef76bebffa2c23#d4cebe383870410cab2e14d0709f78bb).
- If it's a resource identifier, use type [ResourceIdentifier](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/ResourceIdentifier.cs). See example [here](https://apiview.dev/Assemblies/Review/25a98e223bda4f39b4ef76bebffa2c23#9f6fc0ae7c8e497e98d7979dadf97a4c).

File: sdk/workloads/Azure.ResourceManager.Workloads/src/Generated/Models/DiskDetails.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+namespace Azure.ResourceManager.Workloads.Models
+{
+    /// <summary> The supported disk size details for a disk type. </summary>
+    public partial class DiskDetails
```
Comment (archerzz) [inline sdk/workloads/Azure.ResourceManager.Workloads/src/Generated/Models/DiskDetails.cs:11]: The name is too generic. Consider appending RP prefix or renaming it.
- use `prepend-rp-prefix` to automatically append PR name. See example [here](https://github.com/Azure/azure-sdk-for-net/blob/176fd476588e8e61ebce747d66714cb39947cae5/sdk/azurestackhci/Azure.ResourceManager.Hci/src/autorest.md?plain=1#L52-L57).
- use `rename-mapping` to rename a model. See example [here](https://github.com/Azure/azure-sdk-for-net/blob/176fd476588e8e61ebce747d66714cb39947cae5/sdk/azurestackhci/Azure.ResourceManager.Hci/src/autorest.md?plain=1#L58-L76)

Comment (archerzz) [reply sdk/workloads/Azure.ResourceManager.Workloads/src/Generated/Models/DiskDetails.cs:11]: Consider renaming a few related class/enum names, like `DiskSkuName`

File: sdk/workloads/Azure.ResourceManager.Workloads/src/Generated/Models/MountFileShareConfiguration.cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        /// <param name="privateEndpointId"> The private endpoint resource ID. </param>
+        internal MountFileShareConfiguration(ConfigurationType configurationType, string id, string privateEndpointId) : base(configurationType)
+        {
+            Id = id;
+            PrivateEndpointId = privateEndpointId;
+            ConfigurationType = configurationType;
+        }
+
+        /// <summary> The fileshare resource ID. </summary>
+        public string Id { get; set; }
```
Comment (archerzz) [inline sdk/workloads/Azure.ResourceManager.Workloads/src/Generated/Models/MountFileShareConfiguration.cs:42]: What's the format of this property value?
- If it's a GUID, use type [Guid](https://learn.microsoft.com/en-us/dotnet/api/system.guid?view=net-6.0). See example [here](https://apiview.dev/Assemblies/Review/25a98e223bda4f39b4ef76bebffa2c23#d4cebe383870410cab2e14d0709f78bb).
- If it's a resource identifier, use type [ResourceIdentifier](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/ResourceIdentifier.cs). See example [here](https://apiview.dev/Assemblies/Review/25a98e223bda4f39b4ef76bebffa2c23#9f6fc0ae7c8e497e98d7979dadf97a4c).

File: sdk/workloads/Azure.ResourceManager.Workloads/src/Generated/Models/MountFileShareConfiguration.cs [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        /// <summary> The private endpoint resource ID. </summary>
+        public string PrivateEndpointId { get; set; }
```
Comment (archerzz) [inline sdk/workloads/Azure.ResourceManager.Workloads/src/Generated/Models/MountFileShareConfiguration.cs:44]: What's the format of this property value?
- If it's a GUID, use type [Guid](https://learn.microsoft.com/en-us/dotnet/api/system.guid?view=net-6.0). See example [here](https://apiview.dev/Assemblies/Review/25a98e223bda4f39b4ef76bebffa2c23#d4cebe383870410cab2e14d0709f78bb).
- If it's a resource identifier, use type [ResourceIdentifier](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/ResourceIdentifier.cs). See example [here](https://apiview.dev/Assemblies/Review/25a98e223bda4f39b4ef76bebffa2c23#9f6fc0ae7c8e497e98d7979dadf97a4c).

**Review by archerzz — COMMENTED** [review]
Generally looks good.

@ArthurMa1978 Do we allow deleting API in beta releases? Workloads has never published an official release before.

File: sdk/workloads/Azure.ResourceManager.Workloads/src/Generated/SapMonitorData.cs (hunk 1 above)
Comment (gargankit-microsoft) [reply sdk/workloads/Azure.ResourceManager.Workloads/src/Generated/SapMonitorData.cs:91]: https://github.com/Azure/azure-rest-api-specs/blob/main/specification/workloads/resource-manager/Microsoft.Workloads/preview/2022-11-01-preview/monitors.json#L1100

this is a string in swagger, so kept it same here.

File: sdk/workloads/Azure.ResourceManager.Workloads/src/Generated/Models/MountFileShareConfiguration.cs (hunk 3 above)
Comment (gargankit-microsoft) [reply sdk/workloads/Azure.ResourceManager.Workloads/src/Generated/Models/MountFileShareConfiguration.cs:42]: Kept it string, same as swagger: https://github.com/Azure/azure-rest-api-specs/blob/main/specification/workloads/resource-manager/Microsoft.Workloads/preview/2022-11-01-preview/SAPVirtualInstance.json#L2995

File: sdk/workloads/Azure.ResourceManager.Workloads/src/Generated/Models/MountFileShareConfiguration.cs (hunk 4 above)
Comment (gargankit-microsoft) [reply sdk/workloads/Azure.ResourceManager.Workloads/src/Generated/Models/MountFileShareConfiguration.cs:44]: Kept it string, same as swagger: https://github.com/Azure/azure-rest-api-specs/blob/main/specification/workloads/resource-manager/Microsoft.Workloads/preview/2022-11-01-preview/SAPVirtualInstance.json#L2999

File: sdk/workloads/Azure.ResourceManager.Workloads/src/Generated/Models/DiskDetails.cs (hunk 2 above)
Comment (gargankit-microsoft) [reply sdk/workloads/Azure.ResourceManager.Workloads/src/Generated/Models/DiskDetails.cs]: Done

**Review by archerzz — COMMENTED** [review]
Please fix the cspell error: https://dev.azure.com/azure-sdk/public/_build/results?buildId=2604520&view=logs&j=b70e5e73-bbb6-5567-0939-8415943fadb9&t=a880e989-7d1a-5c96-a41f-d540b383cc43&l=122

**Review by archerzz — APPROVED** [review]

**Comment by gargankit-microsoft** [comment]
Moving to another branch due to major conflicts to avoid errors in rebasing: https://github.com/Azure/azure-sdk-for-net/pull/34718

**Pull request merged**